### DISCUSSION TITLE
breaking change: rename fx-form to fx-fore

### DISCUSSION THREAD

**Discussion by JoernT** [Jinntec/Fore#55] [General]
Due to its original orientation as an 'xformish' toolkit it was tempting to name the main element 'fx-form'. 

But even back in the days of the XForms it has been discussed more than once if 'forms' are actually hitting the nature of the spec which does much more than what users usually expect from a form.

As the premises of XForms are still more than valid in these days  Fore has been evolved along those lines without maybe even following each and every detail. But the overall processing model and architecture is very much XForms as it's just a stable MVC design assuring consistency of data all the time. 

But a completely new field has emerged where Fore can shine. While being itself a set of web components it is a nice candidate if you need to assemble more complex pages made of components.

Fore could replace the need for a wrapping component and let components share the data model. As multiple data instances JSON and XML data models can be used and mixed. With events and dispatching of those compents can be wired together. This is a complete new playground that needs exploration :)

To cut things short i decided to rename <fx-form> to <fx-fore> to move the focus away from just pure forms.

**Comment by AlainCouthures** [comment]
I agree with you that the XForms approach is not just about forms. It is also about data rendering, with repeat and output controls and AVT, possibly at client-side (always nice for limited servers).

I would add that it can be an interactive data rendering tool using "form controls" but without any form submission...

It is a "noJS" architecture thanks to actions, XPath power and declarative bindings similar to spreadsheet formulas. As a consequence, it is very concise, which is excellent for prototyping but also for production with good performance.

**Reply by JoernT** [reply]
Agree with you that submissions (at least not for sending data) are not always needed esp. in context with web components that can be nicely configured via bindings. 

The submission module can be used nicely as a declarative variant of the JS fetch API and is certainly much easier to handle for non-programmers to load/send a bunch of data.
